### PR TITLE
feat: parameterize mev-boost args

### PR DESCRIPTION
**Pull request by bharath-123**
Adds 2 new fields to the config file under the mev_params field:
mev_boost_image: <The mev-boost image, default is flashbots/mev-boost>
mev_boost_args: <The args of mev-boost. The default params are ["mev-boost", "--relay-check"]> 

The behaviour is such that, `mev-boost-args` overrides all the args of mev-boost. No extra params are added.

Resolves https://github.com/kurtosis-tech/ethereum-package/issues/289

### CODE REVIEW

**Comment by barnabasbusa** [comment]
Could you please also update the readme?

**Comment by bharath-123** [comment]
will do

**Comment by bharath-123** [comment]
done @barnabasbusa

File: README.md [hunk 1]
```diff
@@ -320,14 +320,16 @@ mev_type: null
...
-  # The image to use for MEV boot relay
+  # The image to use for MEV boost relay
   mev_relay_image: flashbots/mev-boost-relay
   # The image to use for the builder
   mev_builder_image: ethpandaops/flashbots-builder:main
   # The image to use for the CL builder
   mev_builder_cl_image: sigp/lighthouse:latest
   # The image to use for mev-boost
   mev_boost_image: flashbots/mev-boost
+  # Parameters for MEV Boost
```
Comment (h4ck3rk3y) [inline README.md:331]: can you make it clear that this will override completely?

File: main.star [hunk 2]
```diff
@@ -241,9 +241,7 @@ def run(plan, args={}):
     if mev_endpoints:
         for index, participant in enumerate(all_participants):
             if args_with_right_defaults.participants[index].validator_count != 0:
-                mev_boost_launcher = mev_boost.new_mev_boost_launcher(
-                    MEV_BOOST_SHOULD_CHECK_RELAY, mev_endpoints
```
Comment (h4ck3rk3y) [inline main.star:245]: lest delete the constant if it isn't being used?

File: src/package_io/input_parser.star [hunk 3]
```diff
@@ -442,6 +443,7 @@ def get_default_mev_params():
         "mev_builder_image": "flashbots/builder:latest",
         "mev_builder_cl_image": "sigp/lighthouse:latest",
         "mev_boost_image": "flashbots/mev-boost",
+        "mev_boost_args": ["mev-boost", "--relay-check"],
```
Comment (h4ck3rk3y) [inline src/package_io/input_parser.star:446]: why is `mev-boost` an arg to MEV?

Comment (h4ck3rk3y) [reply src/package_io/input_parser.star:446]: i believe `mev-boost` should be the entry point / default command and everything else should be extra

File: README.md (hunk 1 above)
Comment (barnabasbusa) [reply README.md:331]: I think it makes sense to replace the args, as if you use a non flashbot's builder, then you most likely will have completely different flags to define.

**Comment by barnabasbusa** [comment]
@parithosh can you please take a look at this, and ensure this satisfies #289 ?

**Comment by parithosh** [comment]
> @parithosh can you please take a look at this, and ensure this satisfies #289 ?

on it

**Comment by parithosh** [comment]
Hmm, here's an example of a new mev-boost implimentation: https://github.com/mevsharp/mevsharp

To configure it, i can theoretically do this:
```
  mev_boost_image: mevsharp/mevsharp:latest
  mev_boost_extra_args: ["mevsharp","--genesis-fork-version=0x10000038","...."]
```
The issue is that since its a total override, i'd need to know the relay address in advance right? Or is the mev launcher auto-adding it?

I think the only way it can work neatly is if the `mevsharp` team adds config via env vars. Then we can set the env var similar to how it is today: `  "RELAYS": mev_boost_launcher.relay_end_points[0],` to all images and we can override everything with ease. 

Am i missing something or shall i go ahead and make this request of the mevsharp team?

File: README.md (hunk 1 above)
Comment (bharath-123) [reply README.md:331]: yeah also i belive this an important design decision for ethereum-package. Do we want to maintain seperate module for each type of impl or a generic impl but allow to override the args.

File: main.star (hunk 2 above)
Comment (bharath-123) [reply main.star:245]: yes will do

File: src/package_io/input_parser.star (hunk 3 above)
Comment (bharath-123) [reply src/package_io/input_parser.star:446]: it is the case for the flashbots impl. I am unsure about other impls. I ll check out @parithosh mentioned mev-sharp

**Comment by shakeshar** [comment]
> Hmm, here's an example of a new mev-boost implimentation: https://github.com/mevsharp/mevsharp
> 
> To configure it, i can theoretically do this:
> 
> ```
>   mev_boost_image: mevsharp/mevsharp:latest
>   mev_boost_extra_args: ["mevsharp","--genesis-fork-version=0x10000038","...."]
> ```
> 
> The issue is that since its a total override, i'd need to know the relay address in advance right? Or is the mev launcher auto-adding it?
> 
> I think the only way it can work neatly is if the `mevsharp` team adds config via env vars. Then we can set the env var similar to how it is today: ` "RELAYS": mev_boost_launcher.relay_end_points[0],` to all images and we can override everything with ease.
> 
> Am i missing something or shall i go ahead and make this request of the mevsharp team?

We are testing this in the dev branch at the moment. if there are any specific/required variables you need please let me know
Commit: https://github.com/mevsharp/mevsharp/commit/88520c37d344ee9f880e55f2bb35b12f60e143da

**Comment by parithosh** [comment]
> > Hmm, here's an example of a new mev-boost implimentation: https://github.com/mevsharp/mevsharp
> > To configure it, i can theoretically do this:
> > ```
> >   mev_boost_image: mevsharp/mevsharp:latest
> >   mev_boost_extra_args: ["mevsharp","--genesis-fork-version=0x10000038","...."]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The issue is that since its a total override, i'd need to know the relay address in advance right? Or is the mev launcher auto-adding it?
> > I think the only way it can work neatly is if the `mevsharp` team adds config via env vars. Then we can set the env var similar to how it is today: ` "RELAYS": mev_boost_launcher.relay_end_points[0],` to all images and we can override everything with ease.
> > Am i missing something or shall i go ahead and make this request of the mevsharp team?
> 
> We are testing this in the dev branch at the moment. if there are any specific/required variables you need please let me know Commit: [mevsharp/mevsharp@88520c3](https://github.com/mevsharp/mevsharp/commit/88520c37d344ee9f880e55f2bb35b12f60e143da)

Thanks! 

Would be great to support the same ones as flashbots: https://github.com/flashbots/mev-boost/blob/bdabd0e6181990f9c3e0fef9f750b40f71d50c6c/cli/main.go#L35

**Comment by shakeshar** [comment]
> > > Hmm, here's an example of a new mev-boost implimentation: https://github.com/mevsharp/mevsharp
> > > To configure it, i can theoretically do this:
> > > ```
> > >   mev_boost_image: mevsharp/mevsharp:latest
> > >   mev_boost_extra_args: ["mevsharp","--genesis-fork-version=0x10000038","...."]
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The issue is that since its a total override, i'd need to know the relay address in advance right? Or is the mev launcher auto-adding it?
> > > I think the only way it can work neatly is if the `mevsharp` team adds config via env vars. Then we can set the env var similar to how it is today: ` "RELAYS": mev_boost_launcher.relay_end_points[0],` to all images and we can override everything with ease.
> > > Am i missing something or shall i go ahead and make this request of the mevsharp team?
> > 
> > 
> > We are testing this in the dev branch at the moment. if there are any specific/required variables you need please let me know Commit: [mevsharp/mevsharp@88520c3](https://github.com/mevsharp/mevsharp/commit/88520c37d344ee9f880e55f2bb35b12f60e143da)
> 
> Thanks!
> 
> Would be great to support the same ones as flashbots: https://github.com/flashbots/mev-boost/blob/bdabd0e6181990f9c3e0fef9f750b40f71d50c6c/cli/main.go#L35

I followed mev-boost variables so basically Im using the same environment names. Please give it another try and don't hesitate to get back to me if you have any further concerns :)

/Stefan

File: README.md (hunk 1 above)
Comment (h4ck3rk3y) [reply README.md]: @barnabasbusa fair enough! :-)

**Comment by h4ck3rk3y** [comment]
@bharath-123 have you had a chance to revisit this?

**Comment by bharath-123** [comment]
hey @h4ck3rk3y will close it tommorow! thanks for reminding!

**Comment by h4ck3rk3y** [comment]
@parithosh does this look okay to you now?

**Review by parithosh — APPROVED** [review]

**Comment by shakeshar** [comment]
> PR itself seems to work as intended! Thank you @bharath-123 :D
> 
> @shakeshar , seems like mevsharp might need to add support for starting up relying purely on env vars as it currently fails expecting a network flag. But there is no `custom` network flag as far as i can see.
> 
> Minimal reproduction yaml:
> 
> ```
> additional_services: []
> mev_type: mock
> mev_params:
>   mev_boost_image: mevsharp/mevsharp:latest
>   mev_boost_args: ["mevsharp"]
> ```

Yes, that seems like a good idéa.  
At the moment you can set the network using the CLI --network parameter, I could add this as an environment variable, however "network" is quite generic and could be used in other applications I reckon. 
Do you have any preferred variable name you would like to use for this?

I.e in windows this would be set using "setx /M PREFERRED_KEY_NAME sepolia"

**Comment by parithosh** [comment]
> > PR itself seems to work as intended! Thank you @bharath-123 :D
> > @shakeshar , seems like mevsharp might need to add support for starting up relying purely on env vars as it currently fails expecting a network flag. But there is no `custom` network flag as far as i can see.
> > Minimal reproduction yaml:
> > ```
> > additional_services: []
> > mev_type: mock
> > mev_params:
> >   mev_boost_image: mevsharp/mevsharp:latest
> >   mev_boost_args: ["mevsharp"]
> > ```
> 
> Yes, that seems like a good idéa. At the moment you can set the network using the CLI --network parameter, I could add this as an environment variable, however "network" is quite generic and could be used in other applications I reckon. Do you have any preferred variable name you would like to use for this?
> 
> I.e in windows this would be set using "setx /M PREFERRED_KEY_NAME sepolia"

I think the mev-boost behaviour is probably useful to emulate here as well. So if you specify `--network goerli`, then it knows the `GENESIS_FORK_VERSION` value it needs to use for said network. If you want a custom network, you can just forego the `--network` flag and simply set the `GENESIS_FORK_VERSION` directly. 

So the change on your side would be to crash only if `network` and `GENESIS_FORK_VERSION` (or flag variant) is missing.

**Comment by shakeshar** [comment]
Added additional support for environmentvariables. It's been running in Testnet successfully for the past 24/h, get the latest package. 
PR: https://github.com/mevsharp/mevsharp/pull/120
@parithosh

**Pull request merged**